hi-ml/src/health_ml/utils/data_augmentations.py

```diff
@@ -131,14 +131,20 @@ def stain_normalize(img: torch.Tensor, reference_mean: np.ndarray, reference_std
         return nimg
 
     def __call__(self, img: torch.Tensor) -> torch.Tensor:
+        original_shape = img.shape
+        if len(original_shape) == 3:
+            img = img.unsqueeze(0)  # add batch dimension if missing
         # if the input is a bag of images, stain normalization needs to run on each image separately
         if img.shape[0] > 1:
             for i in range(img.shape[0]):
                 img_tile = img[i]
                 img[i] = self.stain_normalize(img_tile.unsqueeze(0), self.reference_mean, self.reference_std)
             return img
         else:
-            return self.stain_normalize(img, self.reference_mean, self.reference_std)
+            img = self.stain_normalize(img, self.reference_mean, self.reference_std)
+            if len(original_shape) == 3:
+                return img.squeeze(0)
```

Comment:
Collate function above expect images (C, H, W) so we need to unsqueeze the image back

Reply:
unsqueeze or squeeze? shouldn't this be `if len(original_shape) == 4` if there is already an existing batch dimension and you want to squeeze?

Reply:
If there's already a batch dimension, I don't want to remove it, I only squeeze when original_shape==3 to put it back to 3 cause I add it line 136